go/cmd/data/cmd/containerwatcher/spns_watcher.go

```diff
@@ -0,0 +1,133 @@
+package containerwatcher
+
+import (
+	"context"
+	"os"
+	"time"
+
+	"github.com/lyft/flytestdlib/logger"
+	"github.com/mitchellh/go-ps"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const (
+	k8sPauseContainerPid = 1
+	k8sAllowedParentPid  = 0
+)
+
+// given list of processes returns a list of processes such that,
+// the pid does not match any of the given filterPids, this is to filter the /pause and current process.
+// and the parentPid is the allowedParentPid. The logic for this is because every process in the shared namespace
+// always has a parent pid of 0
+func FilterProcessList(procs []ps.Process, filterPids sets.Int, allowedParentPid int) ([]ps.Process, error) {
+	var filteredProcs []ps.Process
+	for _, p := range procs {
+		proc := p
+		if proc.PPid() == allowedParentPid {
+			if !filterPids.Has(proc.Pid()) {
+				filteredProcs = append(filteredProcs, proc)
+			}
+		}
+	}
+	return filteredProcs, nil
+}
+
+type SharedNamespaceProcessLister struct {
+	// PID for the current process
+	currentProcessPid int
+	pidsToFilter      sets.Int
+}
+
+func (s *SharedNamespaceProcessLister) AnyProcessRunning(ctx context.Context) (bool, error) {
+	procs, err := s.ListRunningProcesses(ctx)
+	if err != nil {
+		return false, err
+	}
+	return len(procs) > 0, nil
```

Comment:
with "me" as a process, this will always be at least 1, right? shouldn't the check be `> 1` ?

Reply:
no it removes the `/pause` and `self`. Check the documentation on Filter processe